drivers/cache/cache_andes.c

```diff
@@ -0,0 +1,737 @@
+/*
+ * Copyright (c) 2024 Andes Technology Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "soc_v5.h"
+
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/arch/cpu.h>
+#include <zephyr/drivers/syscon.h>
+#include <zephyr/drivers/cache.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/arch/riscv/csr.h>
+
+LOG_MODULE_REGISTER(cache_andes, CONFIG_CACHE_LOG_LEVEL);
+
+/* L1 CCTL Command */
+#define CCTL_L1D_VA_INVAL	0
+#define CCTL_L1D_VA_WB		1
+#define CCTL_L1D_VA_WBINVAL	2
+#define CCTL_L1D_WBINVAL_ALL	6
+#define CCTL_L1D_WB_ALL		7
+#define CCTL_L1I_VA_INVAL	8
+#define CCTL_L1D_INVAL_ALL	23
+#define CCTL_L1I_IX_INVAL	24
+
+#define L2C_EXISTS	DT_NODE_HAS_COMPAT_STATUS(DT_NODE_EXISTS(	\
+			DT_NODELABEL(l2_cache)), andestech_l2c, okay)
+#ifdef L2C_EXISTS
```

Comment:
This will resolve to `true` regardless of the result in L29, should probably be

```suggestion
#if IS_ENABLED(L2C_EXISTS)
```

Reply:
Thanks for catching it up! `DT_NODE_HAS_COMPAT_STATUS` would return 0 or 1, so I think `#if L2C_EXISTS` would be enough.

Reply:
> `#if L2C_EXISTS` would be enough?

yeah that will do as well